### PR TITLE
[CRT] Sync $I10_OUTPUT and test with wine head

### PR DESCRIPTION
**Pull request by tkreuzer**
## Purpose

Fixes $I10_OUTPUT and test on MSVC builds.
Both were broken on MSVC builds, where a long double is 64 bits and not 80 bits like on GCC. The new code works on MSVC builds, too.

Test: https://reactos.org/testman/compare.php?ids=92407,92411,92414,92416

### CODE REVIEW

File: sdk/lib/crt/misc/i10output.c [hunk 1]
```diff
@@ -1,5 +1,140 @@
 #include <precomp.h>
 
+// From Wine msvcrt.h
+typedef struct {ULONG x80[3];} MSVCRT__LDOUBLE; /* Intel 80 bit FP format has sizeof() 12 */
```
Comment (learn-more) [inline sdk/lib/crt/misc/i10output.c:4]: Would it make sense to add a `static_assert` to validate that the type has the expected size?

File: modules/rostests/winetests/msvcrt/misc.c [hunk 2]
```diff
@@ -126,63 +129,62 @@ typedef struct _I10_OUTPUT_test {
...
-    { 0.0, 10, 0, {0, ' ', 1, "0"}, 1, "" },
-    { 1.0, 10, 0, {1, ' ', 1, "1"}, 1, "000000009" },
-    { -1.0, 10, 0, {1, '-', 1, "1"}, 1, "000000009" },
-    { 1.23, 10, 0, {1, ' ', 3, "123"}, 1, "0000009" },
-    { 1e13, 10, 0, {14, ' ', 1, "1"}, 1, "000000009" },
-    { 1e30, 30, 0, {31, ' ', 21, "100000000000000001988"}, 1, "" },
-    { 1e-13, 10, 0, {-12, ' ', 1, "1"}, 1, "000000000" },
-    { 0.25, 10, 0, {0, ' ', 2, "25"}, 1, "00000000" },
-    { 1.0000001, 10, 0, {1, ' ', 8, "10000001"}, 1, "00" },
+    { { 0x00000000, 0x00000000, 0x0000 /* 0.0 */ }, 10, 0, {0, ' ', 1, "0"}, 1, "" },
```
Comment (HBelusca) [inline modules/rostests/winetests/msvcrt/misc.c:132]: Is this basically a fork of the corresponding wine test that is being done here?

File: sdk/lib/crt/misc/i10output.c [hunk 3]
```diff
@@ -24,22 +159,31 @@ struct _I10_OUTPUT_DATA {
  *      Native sets last byte of data->str to '0' or '9', I don't know what
  *      it means. Current implementation sets it always to '0'.
  */
-int CDECL MSVCRT_I10_OUTPUT(_LDOUBLE ld80, int prec, int flag, struct _I10_OUTPUT_DATA *data)
+int CDECL I10_OUTPUT(MSVCRT__LDOUBLE ld80, int prec, int flag, struct _I10_OUTPUT_DATA *data)
```
Comment (HBelusca) [inline sdk/lib/crt/misc/i10output.c:162]: Why the change of the function name? is it to differentiate it with Wine's implementation?

File: sdk/lib/crt/misc/i10output.c [hunk 4]
```diff
@@ -1,5 +1,140 @@
...
+    }
+    if (fp->exp < -(1<<EXP_BITS))
+    {
+        *d = fp->sign * 0.0;
+        return ERANGE;
+    }
+    fp->exp += MANT_BITS - 1;
+
+    /* normalize mantissa */
+    while(fp->m < (ULONGLONG)1 << (MANT_BITS-1))
```
Comment (HBelusca) [inline sdk/lib/crt/misc/i10output.c:65]:
```suggestion
    while (fp->m < (ULONGLONG)1 << (MANT_BITS-1))
```
same for the two others below (lines 70 and 91).

File: sdk/lib/crt/misc/i10output.c (hunk 1 above)
Comment (HBelusca) [reply sdk/lib/crt/misc/i10output.c:4]: Is it necessary to have an `MSVCRT__LDOUBLE` and a separate `_LDOUBLE` structure (stdlib.h)? Is the latter one a "standard" definition that doesn't actually match what MSVC does?

File: modules/rostests/winetests/msvcrt/misc.c (hunk 2 above)
Comment (tkreuzer) [reply modules/rostests/winetests/msvcrt/misc.c:132]: It is the test from current wine head, so yes, kindof a fork. I could sync the entire test, if that is preferred.

File: sdk/lib/crt/misc/i10output.c (hunk 3 above)
Comment (tkreuzer) [reply sdk/lib/crt/misc/i10output.c:162]: Wine is getting rid of these MSVCRT_ prefixes. This is simply what it looks like in current wine head.

File: sdk/lib/crt/misc/i10output.c (hunk 1 above)
Comment (tkreuzer) [reply sdk/lib/crt/misc/i10output.c:4]: It is 3 ULONGs, I think the size should be obvious. t's also only used in this file.
We need this, because the one from stdlib is different.

File: sdk/lib/crt/misc/i10output.c (hunk 4 above)
Comment (tkreuzer) [reply sdk/lib/crt/misc/i10output.c:65]: It's wine code, so I don't see any benefit in changing it, only to apply our coding standards. It's only going to make future syncs harder, or it's simply going to be lost.

File: modules/rostests/winetests/msvcrt/misc.c (hunk 2 above)
Comment (HBelusca) [reply modules/rostests/winetests/msvcrt/misc.c:132]: Fine if it's closer to what Wine nowadays have.

File: sdk/lib/crt/misc/i10output.c (hunk 4 above)
Comment (HBelusca) [reply sdk/lib/crt/misc/i10output.c:65]: ah ok i didn't know.

**Review by HBelusca — APPROVED** [review]
